xbmc/utils/Variant.cpp

```diff
@@ -609,7 +609,7 @@ const CVariant& CVariant::operator[](unsigned int position) const&
 CVariant CVariant::operator[](unsigned int position) &&
 {
   if (m_type == VariantTypeArray && size() > position)
-    return std::move(m_data.array->at(position));
+    return m_data.array->at(position);
```

Comment:
What was the problem here? The intent was to move the `CVariant` at `position` out of the `std::vector` in case `this` is a rvalue.

Reply:
@neo1973 I just sent a followup: https://github.com/xbmc/xbmc/pull/23490

Didn't touch this part because everything is heavily constified, a std::move anethema. Still see a problem that should be addressed in my PR?